drivers/bluetooth/iton_bt.c

```diff
@@ -0,0 +1,250 @@
+        writePinLow(ITON_BT_IRQ_LINE);
+        #endif
+
+        switch (iton_bt_buffer[0]) {
+            case led_state:
+                iton_bt_led_state = iton_bt_buffer[1];
+                break;
+            case notification:
+                switch (iton_bt_buffer[1]) {
+                    case notif_battery:
+                        iton_bt_rx_battery_notif(iton_bt_buffer[2]);
+                        break;
+                    case notif_bluetooth:
+                        iton_bt_rx_bluetooth_notif(iton_bt_buffer[2]);
+                        break;
+                }
+                break;
+        }
+    }
+}
+#endif
+
+
+void iton_bt_data_cb(SPIDriver *spip) {
+    writePinLow(ITON_BT_IRQ_LINE);
+}
+
+/**
+ * Driver Functions
+ */
+void iton_bt_init(void) {
+    setPinOutput(ITON_BT_IRQ_LINE);
+    setPinInput(ITON_BT_INT_LINE);
+
+    writePinLow(ITON_BT_IRQ_LINE);
+
+#if defined(PAL_USE_CALLBACKS) || defined(PAL_USE_WAIT)
+    palSetLineCallback(ITON_BT_INT_LINE, iton_bt_rx_cb, NULL);
+    palEnableLineEvent(ITON_BT_INT_LINE, PAL_EVENT_MODE_BOTH_EDGES);
+#endif
+
+    spiStart(&ITON_BT_SPI_PORT, &iton_bt_spicfg);
+}
+
+void iton_bt_send(uint8_t cmd, uint8_t *data, uint8_t len) {
+    while (readPin(ITON_BT_IRQ_LINE));
+
+    writePinHigh(ITON_BT_IRQ_LINE);
+    iton_bt_buffer[0] = cmd;
+    memcpy(&iton_bt_buffer[1], data, len);
+    spiStartSend(&ITON_BT_SPI_PORT, len + 1, &iton_bt_buffer[0]);
+}
+
+void iton_bt_send2(uint8_t cmd, uint8_t b1, uint8_t b2) {
+    while (readPin(ITON_BT_IRQ_LINE));
+
+    writePinHigh(ITON_BT_IRQ_LINE);
+    iton_bt_buffer[0] = cmd;
+    iton_bt_buffer[1] = b1;
+    iton_bt_buffer[2] = b2;
+
+    spiStartSend(&ITON_BT_SPI_PORT, 3, &iton_bt_buffer[0]);
+}
+
+inline void iton_bt_send_ack(uint8_t b1, uint8_t b2) {
+    writePinHigh(ITON_BT_IRQ_LINE);
+    iton_bt_buffer[0] = control;
+    iton_bt_buffer[1] = b1;
+    iton_bt_buffer[2] = b2;
+    chSysLockFromISR();
+    spiStartSendI(&ITON_BT_SPI_PORT, 3, &iton_bt_buffer[0]);
+    chSysUnlockFromISR();
+}
+
+void iton_bt_send_fn(bool pressed) {
+    uint8_t data = pressed ? 0xA3 : 0x00;
+
+    iton_bt_send(report_fn, &data, 1);
+}
+
+void iton_bt_send_system(uint16_t data) {
+    iton_bt_send(report_system, (uint8_t *)&data, 1);
+}
+
+void iton_bt_send_consumer(uint16_t data) {
+    iton_bt_send2(report_consumer, HIGH_BITS(data), LOW_BITS(data));
+}
+
+void iton_bt_send_keyboard(report_keyboard_t *report) {
+    // iton module only reads 5 of the keys in the hid report
+    // so we send the last key as an nkro report.
+    if (report->keys[5] != iton_bt_send_kb_last_key) {
+        uint8_t nkro_report[15] = {0};
+        nkro_report[report->keys[5] >> 3] |= (1 << (report->keys[5] & 7));
+        iton_bt_send_kb_last_key = report->keys[5];
+
+        return iton_bt_send(report_nkro, &nkro_report[0], 15);
+    }
+
+    iton_bt_send(report_hid, &report->raw[HID_REPORT_OFFSET], 8);
```

Comment:
I had to change this to:
```
iton_bt_send(report_hid, &report->mods, 8);
```

It seems to be working but I am not sure it was the right thing to do :)